### DISCUSSION TITLE
ChoiceType class Form

### DISCUSSION THREAD

**Discussion by cube4d** [symfony/symfony#45958] [Q&A] [answered]
Hi,

i have a form with an ChoiceType that includes 2 choices.
I like to "style" the 2 labels for the choices itself, but i dont know how.
With the row_attr i change the whole choices field, and i cant find any answer in the docs.

```php
->add('dringlichkeit', ChoiceType::class, [
                'choices'  => [
                    'CHOICE1' => 'gelb',
                    'CHOICE2' => 'rot',
                ],
                'mapped' => false,
                'label' => 'Wie dringend muss der Mangel beseitigt werden?',
                'help' => 'z.B. Reifenschaden, defektes Rücklicht, alle Sicherheits Beeinträchtigungen',
                'expanded' => true,
                'multiple' => false,
                'row_attr' => ['class' => 'text-danger'],
            ])
```

i want that CHOICE1 is yellow, and CHOICE2 is red.
I tried this: 

```php
                'choice_attr' => [
                    'CHOICE1' => ['class' => 'text-danger mt-3'],
                    'CHOICE2' => ['class' => 'text-danger mt-3'],
                   
                ],
```

but that only changed the position of the radio buttons. 
With the 'choice_label' attribute i can change only the text if i understand it right ?

thanks

**Comment by MichaelBrauner** [comment]
What is actually the problem? The `choice_attr` correctly set the class atributes on the radio buttons. Is that right?

**Reply by stof** [reply]
well, choice_attr allows to configure the `attr` option of the child RadioType. But we don't have a `choice_row_attr`.

**Reply by MichaelBrauner** [reply]
I did test it now insite a local installation. 

That's the result so far.

```html
<div class="text-danger">
    <label class="required">Wie dringend muss der Mangel beseitigt werden?</label>

    <div id="profile_dringlichkeit" aria-describedby="profile_dringlichkeit_help">
        <input type="radio" id="profile_dringlichkeit_0" name="profile[dringlichkeit]" required="required" class="text-danger mt-3" value="gelb">
        <label for="profile_dringlichkeit_0" class="required">CHOICE1</label>
        <input type="radio" id="profile_dringlichkeit_1" name="profile[dringlichkeit]" required="required" class="text-danger mt-3" value="rot">
        <label for="profile_dringlichkeit_1" class="required">CHOICE2</label>
    </div>

    <p id="profile_dringlichkeit_help" class="help-text">z.B. Reifenschaden, defektes Rücklicht, alle Sicherheits Beeinträchtigungen</p>
</div>
```

@stof you are right - some kind of `choice_row_attr` or '`choice_label_attr'` might be missing.

I tried to create a FormTypeExtension and a custom form_theme just to realize that options are merged in [createChoiceListView function](https://github.com/symfony/form/blob/000f3e104dfd66a39c9fc5c925f4dfdfbd2010c2/Extension/Core/Type/ChoiceType.php#L489). 

So I guess some work on [DefaultChoiceListFactory](https://github.com/symfony/form/blob/5.4/ChoiceList/Factory/DefaultChoiceListFactory.php) is necessary. 

If you want I can work on that. But I will need some help I guess.

**Comment by MichaelBrauner** [comment, marked as answer]
As a fast solution: 

```php
 'choice_attr' => [
    'CHOICE1' => ['class' => 'choice1'],
    'CHOICE2' => ['class' => 'choice2'],
],
```

And than with css: 

```css
input[class=choice1] + label {
    color: #00FF00;
}
```

I tested it. That works fine.

**Reply by cube4d** [reply]
Ok thanks i will test it :-)

**Comment by MichaelBrauner** [comment]
Ok. I'm glad that I could help. 
Please don't forget to close this discussion afterwards.

**Reply by cube4d** [reply]
I solved it now with CSS yes , thanks for your help :-)
